e2e/utils.go

```diff
@@ -508,6 +508,60 @@ func validatePVCAndAppBinding(pvcPath, appPath string, f *framework.Framework) e
 	return err
 }
 
+// validatePVCAndAppWaitForFirstConsumer creates a Pending PVC, starts an app, and verifies it to become Bound.
+func validatePVCAndAppWaitForFirstConsumer(pvcPath, appPath string, f *framework.Framework) error {
+	pvc, err := loadPVC(pvcPath)
+	if err != nil {
+		return err
+	}
+	pvc.Namespace = f.UniqueName
+
+	app, err := loadApp(appPath)
+	if err != nil {
+		return err
+	}
+	app.Namespace = f.UniqueName
+
+	err = createPVC(f.ClientSet, pvc)
+	if err != nil {
+		return err
+	}
+
+	err = waitForPVCPhase(f.ClientSet, pvc.Namespace, pvc.Name, v1.ClaimPending, 30*time.Second)
+	if err != nil {
+		return fmt.Errorf("PVC did not stay in Pending state as expected: %w", err)
+	}
+
+	err = createApp(f.ClientSet, app, deployTimeout)
+	if err != nil {
+		return err
+	}
+
+	err = waitForPVCPhase(f.ClientSet, pvc.Namespace, pvc.Name, v1.ClaimBound, 30*time.Second)
+	if err != nil {
+		return fmt.Errorf("PVC did not reach Bound state after creating the pod: %w", err)
+	}
```

Comment:
These seems redundant because CreateApp check for the Pod to be in running state which means the PVC is created and mounted to the application.